src/audio2text/utils.py

```diff
@@ -0,0 +1,99 @@
+    """Loads test dataset from GOLOS and saves audio samples to files."""
+    ds = load_dataset("bond005/sberdevices_golos_10h_crowd")
+    data_ = ds["test"]
+    real_transcriptions = []
+    for i, smpl in tqdm.tqdm(enumerate(data_)):
+        filename = os.path.join(path_to_audio, f"output_{i}.wav")
+        write(filename, smpl["audio"]["sampling_rate"], smpl["audio"]["array"])
+        real_transcriptions.append(smpl["transcription"])
+        # Audio(smpl["audio"]["array"], rate=smpl["audio"]["sampling_rate"])
+    return real_transcriptions
+
+
+def gen_transcriptions_golos(
+    model: Wav2Vec2ForCTC,
+    processor: Wav2Vec2Processor,
+    num_samples: int = 10,
+    folder_to_audio: str = "audio_files",
+    whisper_flag: bool = False,
+) -> list[str]:
+    """Generates transcriptions for stored audio files using a speech model."""
+    transcriptions = []
+    audio_paths = [os.path.join(folder_to_audio, f"output_{i}.wav") for i in range(num_samples)]
+    for audio_path in audio_paths:
+        if whisper_flag:
+            transcription = gen_transcription_whisper(model, processor, audio_path)
+        else:
+            transcription = gen_transcription(model, processor, audio_path)
+        transcriptions.append(transcription)
+    return transcriptions
+
+
+def calc_test_metrics(real_transcriptions: list[str], generated_transcriptions: list[str]) -> None:
+    """Calculates Word Error Rate (WER) and Character Error Rate (CER)."""
+    nan_indices = [
+        i for i, val in enumerate(real_transcriptions) if val is None or (isinstance(val, float) and np.isnan(val))
+    ]
+    generated_transcriptions = [val for i, val in enumerate(generated_transcriptions) if i not in nan_indices]
+    real_transcriptions = [val for i, val in enumerate(real_transcriptions) if i not in nan_indices]
+    len_ = len(generated_transcriptions)  # len(generated_transcriptions)
+    print("WER:", wer(real_transcriptions[:len_], generated_transcriptions[:len_]))
+    print("CER:", cer(real_transcriptions[:len_], generated_transcriptions[:len_]))
+
+
+def gen_transcription(model: Wav2Vec2ForCTC, processor: Wav2Vec2Processor, path_to_file: str) -> str:
+    """Generates a transcription for a single audio file."""
+    try:
+        speech_array, _ = librosa.load(path_to_file, sr=16_000)
+        inputs = processor(speech_array, sampling_rate=16_000, return_tensors="pt", padding=True)
+        with torch.no_grad():
+            logits = model(inputs.input_values, attention_mask=inputs.attention_mask).logits
+        predicted_ids = torch.argmax(logits, dim=-1)
+        decoded = processor.batch_decode(predicted_ids)
+        transcription = cast(str, decoded[0]) if decoded else ""
+        return transcription
+    except FileNotFoundError as exc:
+        raise FileNotFoundError(f"No file {path_to_file}") from exc
+    except Exception as e:
+        raise RuntimeError(f"Error during transcription: {e}") from e
```

Comment:
Looks like this try except construction could be removed without influence on code